src/coreclr/vm/comutilnative.cpp

```diff
@@ -1898,46 +1898,36 @@ static bool HasOverriddenStreamMethod(MethodTable * pMT, WORD slot)
     return true;
 }
 
-FCIMPL1(FC_BOOL_RET, StreamNative::HasOverriddenBeginEndRead, Object *stream)
+extern "C" BOOL QCALLTYPE StreamNative_HasOverriddenSlow(MethodTable* pMT, BOOL isRead)
 {
-    FCALL_CONTRACT;
+    QCALL_CONTRACT;
+    _ASSERTE(pMT != NULL);
 
-    if (stream == NULL)
-        FC_RETURN_BOOL(TRUE);
+    BOOL readOverride = FALSE;
+    BOOL writeOverride = FALSE;
 
-    if (g_pStreamMT == NULL || g_slotBeginRead == 0 || g_slotEndRead == 0)
+    BEGIN_QCALL;
+
+    // Check for needed details
+    if (g_pStreamMT == NULL
+        || g_slotBeginRead == 0
+        || g_slotEndRead == 0
+        || g_slotBeginWrite == 0
+        || g_slotEndWrite == 0)
     {
-        HELPER_METHOD_FRAME_BEGIN_RET_1(stream);
         g_pStreamMT = CoreLibBinder::GetClass(CLASS__STREAM);
```

Comment:
I do not think we need to cache the StreamMT and the slots in statics now that this whole thing is a slow path.